src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,44 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Track;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl implements MachineService<Machine> {
+    @Override
+    public List<Machine> getAll(Class<? extends Machine> type) {
+        List<? extends Machine> machines = null;
+        if (type.equals(Bulldozer.class)) {
+            machines = new BulldozerProducer().get();
+            return (List<Machine>) machines;
+        }
+        if (type.equals(Excavator.class)) {
+            machines = new ExcavatorProducer().get();
+            return (List<Machine>) machines;
+        }
+        if (type.equals(Track.class)) {
+            machines = new TrackProducer().get();
+            return (List<Machine>) machines;
```

Comment:
do we need to use return at all `if` statement? 
can we use it only at the end of the method?

Reply:
done